src/DockWidgetSideTab.h

```diff
@@ -0,0 +1,168 @@
+#ifndef DockWidgetSideTabH
+#define DockWidgetSideTabH
+/*******************************************************************************
+** Qt Advanced Docking System
+** Copyright (C) 2017 Uwe Kindler
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License as published by the Free Software Foundation; either
+** version 2.1 of the License, or (at your option) any later version.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+** Lesser General Public License for more details.
+**
+** You should have received a copy of the GNU Lesser General Public
+** License along with this library; If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+
+//============================================================================
+/// \file   DockWidgetTab.h
+/// \author Syarif Fakhri
+/// \date   05.09.2022
+/// \brief  Declaration of CDockWidgetSideTab class
+//============================================================================
+
+//============================================================================
+//                                   INCLUDES
+//============================================================================
+#include <QFrame>
+
+#include "ads_globals.h"
+
+namespace ads
+{
+struct DockWidgetSideTabPrivate;
+class CDockWidget;
+class CSideTabBar;
+class CDockWidgetTab;
+struct SideTabIconLabelPrivate;
+
+/**
+ * A dock widget Side tab that shows a title or an icon.
+ * The dock widget tab is shown in the side tab bar to switch between
+ * pinned dock widgets
+ */
+class ADS_EXPORT CDockWidgetSideTab : public QFrame
+{
```

Comment:
It would be great, if the side tab would be more similar to the CDockWidgetTab - that means, that it has an isActiveTab() function and activeTab property. That means, a side tab bar may have one single active tab or no active tab. This would allow a styling of the tab depending on the visible state of the auto hide widget. 

So a tab would be more like a checkable action. If it is pressed (active) the auto hide dock widget is visible, if not, it is invisible.

Reply:
![image](https://user-images.githubusercontent.com/4475295/195859810-bf3ba8ff-c1a9-42e7-9c00-ff8e8bf9c0c7.png)

Added! Also updated the styling so it's based on isActiveTab. I suppose you wanted to add this to remove the dependency on focusHighlighting?